lib/runner.js

```diff
@@ -347,7 +348,7 @@ class Runner extends Emittery {
 			}
 		}
 
-		const alwaysOk = await this.runHooks(this.tasks.afterEachAlways, contextRef, hookSuffix);
+		const alwaysOk = await this.runHooks(this.tasks.afterEachAlways, contextRef, hookSuffix, hooksAndTestOk);
```

Comment:
This means that, if a `afterEach` hook fails, the `afterEach.always` hook sees `t.passed === false`, even if the test passed.

I think the semantics are clearer if `t.passed` is `true` only when the test passed. If the test didn't run, because a `beforeEach` hook failed, then `t.passed === false`, but if it succeeded and then an `afterEach` hook failed, it should be `t.passed === true`.

What do you think?

Reply:
This is a entitled issue. I fixed it quite now.